### PR TITLE
SHYD-4241 Variable queues support

### PR DESCRIPTION
**Pull request by keon94**

### CODE REVIEW

**Review by BrandonReno1 — APPROVED** [review]

File: worker.go [hunk 1]
```diff
@@ -210,3 +208,15 @@ func runTaskFunc(taskFunc *reflect.Value, message *TaskMessage) (*ResultMessage,
 	}
 	return getReflectionResultMessage(&res[0]), nil
 }
+
+func (w *CeleryWorker) getRegisteredQueues() []string {
+	queuesMap := map[string]interface{}{}
+	for _, v := range w.registeredTasks {
+		queuesMap[v.Queue] = nil
```
Comment (pnongah) [inline worker.go:215]: Why are we setting this to nil here?

File: amqp_broker.go [hunk 2]
```diff
@@ -10,98 +10,90 @@ import (
...
 	}
 	defer func() {
 		err = channel.Close()
 		if err != nil {
-			fmt.Println("error closing channel")
+			logrus.Warnf("error closing channel")
 		}
 	}()
-	taskMessage := message.GetTaskMessage()
+	queue := b.defaultSendQueue
```
Comment (pnongah) [inline amqp_broker.go:90]: If we are always sending to the defaultSendQueue, but we can [listen on several queues](https://github.com/pnongah/gocelery/blob/b2b718941bf426e47d3db60e28655eef2f1142bf/amqp_broker.go#L61) based on the task;  how do we get a message to a particular task on a particular queue?

**Review by pnongah — CHANGES_REQUESTED** [review]

File: worker.go (hunk 1 above)
Comment (keon94) [reply worker.go:215]: I'm using the map as a set here, so the values don't matter

File: amqp_broker.go (hunk 2 above)
Comment (keon94) [reply amqp_broker.go:90]: we don't always send on the default. It's adjustable at the Delay() API call level. Look at the fields on the gocelery.TaskParameters struct: it has a Queue field which the consumer can adjust on a per-function call basis. If not specified, we fall back to this one.

**Pull request merged**